src/board/novacustom/v540tnx/gpio.c

```diff
@@ -0,0 +1,266 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+#include <board/gpio.h>
+#include <common/macro.h>
+
+// clang-format off
+struct Gpio __code ACIN_N =         GPIO(B, 0);
+struct Gpio __code AC_PRESENT =     GPIO(F, 7);
+struct Gpio __code ALL_SYS_PWRGD =  GPIO(C, 0);
+struct Gpio __code BKL_EN =         GPIO(C, 7);
+struct Gpio __code BUF_PLT_RST_N =  GPIO(D, 2);
+struct Gpio __code CCD_EN =         GPIO(D, 1);
+struct Gpio __code DD_ON =          GPIO(E, 4);
+struct Gpio __code DGPU_PWR_EN =    GPIO(H, 4);
+struct Gpio __code EC_EN =          GPIO(B, 6);
+struct Gpio __code EC_RSMRST_N =    GPIO(E, 5);
+struct Gpio __code GC6_FB_EN =      GPIO(J, 3);
+struct Gpio __code H_PROCHOT_EC =   GPIO(G, 6);
+struct Gpio __code JACK_IN_N =      GPIO(C, 6);
+struct Gpio __code LAN_WAKEUP_N =   GPIO(B, 2);
+struct Gpio __code LED_ACIN =       GPIO(H, 2);
+struct Gpio __code LED_BAT_CHG =    GPIO(H, 5);
+struct Gpio __code LED_BAT_FULL =   GPIO(J, 0);
+struct Gpio __code LED_PWR =        GPIO(D, 0);
+struct Gpio __code LID_SW_N =       GPIO(B, 1);
+struct Gpio __code PCH_PWROK_EC =   GPIO(F, 3);
+struct Gpio __code PD_EN =          GPIO(D, 4);
+struct Gpio __code PWR_BTN_N =      GPIO(D, 5);
+struct Gpio __code PWR_SW_N =       GPIO(B, 3);
+struct Gpio __code RGBKB_DET_N =    GPIO(I, 2);
+struct Gpio __code SINK_CTRL =      GPIO(H, 7);
+struct Gpio __code SUSB_N_PCH =     GPIO(H, 0);
+struct Gpio __code SUSC_N_PCH =     GPIO(H, 1);
+struct Gpio __code VA_EC_EN =       GPIO(J, 4);
+struct Gpio __code WLAN_PWR_EN =    GPIO(E, 1);
+struct Gpio __code XLP_OUT =        GPIO(B, 4);
+// clang-format on
+
+void gpio_init(void) {
+    // Set global configuration
+    // Enable LPC reset on GPD2
+    GCR = 0b10 << 1;
+    // Disable UARTs
+    GCR6 = 0;
+    // PWRSW WDT 2 Enable 1
+    GCR8 = BIT(4);
+    // PWRSW WDT 2 Enable 2
+    GCR9 = BIT(5);
+    // Enable SMBus channel 4
+    GCR15 = BIT(4);
+    // Set GPB5 and GPD2 to 1.8V
+    GCR19 = BIT(7) | BIT(0);
+    // Set GPD3 to 1.8V, GPF2 and GPF3 to 3.3V
```

Comment:
Schematics says GPF2 and GPF3 is 1.8V

Reply:
GPF3 thankfully goes to an U56 which accepts 1.2 ~ 5V, but yeah this probably needs to be fixed.